// ngx-pretty-checkbox/esm5/lib/components/ngx-pretty-Indeterminate.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// tslint:disable:use-host-property-decorator
import { Component, ElementRef, Renderer2, ChangeDetectionStrategy, Attribute } from '@angular/core';
import { DEFAULT_PREFIX, DEFAULT_OUTLINE_PREFIX } from '../model/params';
import { PrettyCheckboxColor } from '../model/interfaces';
var NgxPrettyIndeterminateComponent = /** @class */ (function () {
    function NgxPrettyIndeterminateComponent(el, renderer, _outline, _color) {
        if (_outline === void 0) { _outline = false; }
        this.el = el;
        this.renderer = renderer;
        this._outline = _outline;
        this._color = _color;
    }
    /**
     * @return {?}
     */
    NgxPrettyIndeterminateComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        if (this._color) {
            this.renderer.addClass(this.el.nativeElement, "" + DEFAULT_PREFIX + this._color + (this._outline ? DEFAULT_OUTLINE_PREFIX : ''));
        }
    };
    NgxPrettyIndeterminateComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-p-indeterminate:not([will-change]), p-indeterminate:not([will-change])',
                    template: "\n    <ng-content select=\"[pIcon], [p-icon], [pSvg], [p-svg], [pImage], [p-image]\"></ng-content>\n    <label><ng-content></ng-content></label>\n  ",
                    host: {
                        class: "state " + DEFAULT_PREFIX + "is-indeterminate"
                    },
                    changeDetection: ChangeDetectionStrategy.OnPush
                }] }
    ];
    /** @nocollapse */
    NgxPrettyIndeterminateComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 },
        { type: undefined, decorators: [{ type: Attribute, args: ['outline',] }] },
        { type: PrettyCheckboxColor, decorators: [{ type: Attribute, args: ['color',] }] }
    ]; };
    return NgxPrettyIndeterminateComponent;
}());
export { NgxPrettyIndeterminateComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxPrettyIndeterminateComponent.prototype.el;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyIndeterminateComponent.prototype.renderer;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyIndeterminateComponent.prototype._outline;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyIndeterminateComponent.prototype._color;
}
