// ngx-pretty-checkbox/esm5/lib/components/ngx-pretty-toggle.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// tslint:disable:use-host-property-decorator
import { ElementRef, Component, Renderer2, ChangeDetectionStrategy, Attribute } from '@angular/core';
import { PrettyCheckboxColor, PrettyCheckBoxToggleType } from '../model/interfaces';
import { DEFAULT_PREFIX, DEFAULT_OUTLINE_PREFIX } from '../model/params';
var NgxPrettyToggleComponent = /** @class */ (function () {
    function NgxPrettyToggleComponent(el, renderer, _outline, _color, _type) {
        if (_outline === void 0) { _outline = false; }
        this.el = el;
        this.renderer = renderer;
        this._outline = _outline;
        this._color = _color;
        this._type = _type;
    }
    /**
     * @return {?}
     */
    NgxPrettyToggleComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        if (this._color) {
            this.renderer.addClass(this.el.nativeElement, "" + DEFAULT_PREFIX + this._color + (this._outline ? DEFAULT_OUTLINE_PREFIX : ''));
        }
        this.renderer.addClass(this.el.nativeElement, "" + DEFAULT_PREFIX + this._type);
    };
    NgxPrettyToggleComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-p-toggle:not([will-change]), p-toggle:not([will-change])',
                    template: "\n    <ng-content select=\"[pIcon], [p-icon], [pSvg], [p-svg], [pImage], [p-image]\"></ng-content>\n    <label><ng-content></ng-content></label>\n  ",
                    host: {
                        class: 'state'
                    },
                    changeDetection: ChangeDetectionStrategy.OnPush
                }] }
    ];
    /** @nocollapse */
    NgxPrettyToggleComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 },
        { type: undefined, decorators: [{ type: Attribute, args: ['outline',] }] },
        { type: PrettyCheckboxColor, decorators: [{ type: Attribute, args: ['color',] }] },
        { type: PrettyCheckBoxToggleType, decorators: [{ type: Attribute, args: ['type',] }] }
    ]; };
    return NgxPrettyToggleComponent;
}());
export { NgxPrettyToggleComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleComponent.prototype.el;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleComponent.prototype.renderer;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleComponent.prototype._outline;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleComponent.prototype._color;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleComponent.prototype._type;
}
