// ngx-pretty-checkbox/esm5/lib/components/ngx-pretty-toggle-will-change.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// tslint:disable:use-host-property-decorator
import { Input, ElementRef, Component, HostBinding, Renderer2, ChangeDetectionStrategy } from '@angular/core';
import { PrettyCheckboxColor, PrettyCheckBoxToggleType } from '../model/interfaces';
import { DEFAULT_PREFIX } from '../model/params';
import { getColorClassName } from '../utility';
var NgxPrettyToggleWillChangeComponent = /** @class */ (function () {
    function NgxPrettyToggleWillChangeComponent(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.outline = false;
    }
    Object.defineProperty(NgxPrettyToggleWillChangeComponent.prototype, "isToggleOn", {
        get: /**
         * @return {?}
         */
        function () { return this.type === PrettyCheckBoxToggleType.On; },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NgxPrettyToggleWillChangeComponent.prototype, "isToggleOff", {
        get: /**
         * @return {?}
         */
        function () { return this.type === PrettyCheckBoxToggleType.Off; },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} changes
     * @return {?}
     */
    NgxPrettyToggleWillChangeComponent.prototype.ngOnChanges = /**
     * @param {?} changes
     * @return {?}
     */
    function (changes) {
        if (changes.color || changes.outline) {
            /** @type {?} */
            var oldClass = getColorClassName(changes.color ? changes.color.previousValue : this.color, changes.outline ? changes.outline.previousValue : null);
            /** @type {?} */
            var newClass = getColorClassName(changes.color ? changes.color.currentValue : this.color, changes.outline ? changes.outline.currentValue : null);
            this.renderer.removeClass(this.el.nativeElement, oldClass);
            this.renderer.addClass(this.el.nativeElement, newClass);
        }
    };
    NgxPrettyToggleWillChangeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-p-toggle[will-change], p-toggle[will-change]',
                    template: "\n    <ng-content select=\"[pIcon], [p-icon], [pSvg], [p-svg], [pImage], [p-image]\"></ng-content>\n    <label><ng-content></ng-content></label>\n  ",
                    host: {
                        class: 'state'
                    },
                    changeDetection: ChangeDetectionStrategy.OnPush
                }] }
    ];
    /** @nocollapse */
    NgxPrettyToggleWillChangeComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    NgxPrettyToggleWillChangeComponent.propDecorators = {
        type: [{ type: Input }],
        isToggleOn: [{ type: HostBinding, args: ["class." + DEFAULT_PREFIX + "on",] }],
        isToggleOff: [{ type: HostBinding, args: ["class." + DEFAULT_PREFIX + "off",] }],
        color: [{ type: Input }],
        outline: [{ type: Input }]
    };
    return NgxPrettyToggleWillChangeComponent;
}());
export { NgxPrettyToggleWillChangeComponent };
if (false) {
    /** @type {?} */
    NgxPrettyToggleWillChangeComponent.prototype.type;
    /** @type {?} */
    NgxPrettyToggleWillChangeComponent.prototype.color;
    /** @type {?} */
    NgxPrettyToggleWillChangeComponent.prototype.outline;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleWillChangeComponent.prototype.el;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleWillChangeComponent.prototype.renderer;
}
