// ngx-pretty-checkbox/esm2015/lib/model/interfaces.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @enum {string} */
const PrettyCheckBoxToggleType = {
    On: 'on',
    Off: 'off',
};
export { PrettyCheckBoxToggleType };
/** @enum {string} */
const PrettyCheckboxStroke = {
    // Accept Null
    Fill: 'fill',
    Thick: 'thick',
    Slim: 'slim' // Switch
    ,
};
export { PrettyCheckboxStroke };
/** @enum {string} */
const PrettyCheckboxShape = {
    // Accept Null
    Curve: 'curve',
    Round: 'round',
};
export { PrettyCheckboxShape };
/** @enum {string} */
const PrettyCheckboxColor = {
    Primary: 'primary',
    Success: 'success',
    Info: 'info',
    Warning: 'warning',
    Danger: 'danger',
};
export { PrettyCheckboxColor };
/** @enum {string} */
const PrettyCheckboxAnimation = {
    Smooth: 'smooth',
    Jelly: 'jelly',
    Tada: 'tada',
    Rotate: 'rotate',
    Pulse: 'pulse',
};
export { PrettyCheckboxAnimation };
/**
 * @record
 */
export function PrettyCheckBoxChange() { }
if (false) {
    /** @type {?} */
    PrettyCheckBoxChange.prototype.checked;
    /** @type {?} */
    PrettyCheckBoxChange.prototype.value;
    /** @type {?} */
    PrettyCheckBoxChange.prototype.event;
}
/**
 * @record
 */
export function PrettyRadioChange() { }
if (false) {
    /** @type {?} */
    PrettyRadioChange.prototype.checked;
    /** @type {?} */
    PrettyRadioChange.prototype.value;
    /** @type {?} */
    PrettyRadioChange.prototype.event;
}
