// ngx-pretty-checkbox/esm2015/lib/components/ngx-pretty-radio-group.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, Output, EventEmitter } from '@angular/core';
import { NgxPrettyRadioService } from '../ngx-pretty-checkbox.service';
import { GROUP_NAME } from '../model/params';
/** @type {?} */
let nextId = 1;
export class NgxPrettyRadioGroupDirective {
    constructor() {
        this.name = GROUP_NAME + nextId++;
        this.change = new EventEmitter();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    _emitChange(event) {
        this.change.emit(event);
    }
}
NgxPrettyRadioGroupDirective.decorators = [
    { type: Directive, args: [{
                selector: 'ngx-pretty-radio-group, p-radio-group',
                providers: [NgxPrettyRadioService],
                exportAs: 'ngxRadioGroup'
            },] }
];
/** @nocollapse */
NgxPrettyRadioGroupDirective.ctorParameters = () => [];
NgxPrettyRadioGroupDirective.propDecorators = {
    change: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    NgxPrettyRadioGroupDirective.prototype.name;
    /** @type {?} */
    NgxPrettyRadioGroupDirective.prototype.change;
}
