// ngx-pretty-checkbox/esm2015/lib/components/ngx-pretty-toggle-will-change.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
// tslint:disable:use-host-property-decorator
import { Input, ElementRef, Component, HostBinding, Renderer2, ChangeDetectionStrategy } from '@angular/core';
import { PrettyCheckboxColor, PrettyCheckBoxToggleType } from '../model/interfaces';
import { DEFAULT_PREFIX } from '../model/params';
import { getColorClassName } from '../utility';
export class NgxPrettyToggleWillChangeComponent {
    /**
     * @param {?} el
     * @param {?} renderer
     */
    constructor(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.outline = false;
    }
    /**
     * @return {?}
     */
    get isToggleOn() { return this.type === PrettyCheckBoxToggleType.On; }
    /**
     * @return {?}
     */
    get isToggleOff() { return this.type === PrettyCheckBoxToggleType.Off; }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        if (changes.color || changes.outline) {
            /** @type {?} */
            const oldClass = getColorClassName(changes.color ? changes.color.previousValue : this.color, changes.outline ? changes.outline.previousValue : null);
            /** @type {?} */
            const newClass = getColorClassName(changes.color ? changes.color.currentValue : this.color, changes.outline ? changes.outline.currentValue : null);
            this.renderer.removeClass(this.el.nativeElement, oldClass);
            this.renderer.addClass(this.el.nativeElement, newClass);
        }
    }
}
NgxPrettyToggleWillChangeComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-p-toggle[will-change], p-toggle[will-change]',
                template: `
    <ng-content select="[pIcon], [p-icon], [pSvg], [p-svg], [pImage], [p-image]"></ng-content>
    <label><ng-content></ng-content></label>
  `,
                host: {
                    class: 'state'
                },
                changeDetection: ChangeDetectionStrategy.OnPush
            }] }
];
/** @nocollapse */
NgxPrettyToggleWillChangeComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
NgxPrettyToggleWillChangeComponent.propDecorators = {
    type: [{ type: Input }],
    isToggleOn: [{ type: HostBinding, args: [`class.${DEFAULT_PREFIX}on`,] }],
    isToggleOff: [{ type: HostBinding, args: [`class.${DEFAULT_PREFIX}off`,] }],
    color: [{ type: Input }],
    outline: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    NgxPrettyToggleWillChangeComponent.prototype.type;
    /** @type {?} */
    NgxPrettyToggleWillChangeComponent.prototype.color;
    /** @type {?} */
    NgxPrettyToggleWillChangeComponent.prototype.outline;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleWillChangeComponent.prototype.el;
    /**
     * @type {?}
     * @private
     */
    NgxPrettyToggleWillChangeComponent.prototype.renderer;
}
